File: ООП/Homework2/all_animals/dog.java
package all_animals;

public class dog extends pets{

    private final String train; 

    protected dog(String name, String breed, String vaccination, String color, String Bday, int height, int weight, String eyes_color, String train) {
        super(name, breed, vaccination, Bday, height, weight, eyes_color);
        this.train = train;
    }

    public void training(){
        System.out.println("\nУчится команде 'сидеть' ");
    }

    public void petme() {
        System.out.println("\nПоиграй с собачкой ^_^");
    }

    public void voice() {
        System.out.println("\nГав");
    }

    public String toString(){
        return String.format("Собака  %s\nДрессировка:%s", super.toString(), this.train);
    }
}

File: ООП/Homework2/animals_PY/classes/pets/dog.py
from ..interfaces.training import training 
from .pet import pet

class dog(pet):
    def __init__(self, name, height, weight, coloreye, nickname, breed, vaccination, wool, bday):
        super().__init__(name, height, weight, coloreye, nickname, breed, vaccination, wool, bday)

    def voice(self):
       print(f"{super().voice()} Гав-гав")

    def pet_me(self):
        return super().pet_me()
    
    def training(self):
        training.trainig(self)
    
    def __str__(self) -> str:
        return (f"{super().__str__()}")